packages/grid/src/styles/vaadin-grid-base-styles.js

```diff
@@ -240,14 +246,29 @@ export const gridStyles = css`
   }
 
   /* Variant: row stripes */
-  :host([theme~='row-stripes']) [part~='odd-row'],
-  :host([theme~='row-stripes']) [part~='odd-row'] [part~='cell'] {
-    --vaadin-grid-cell-background: var(
-      --vaadin-grid-odd-row-cell-background,
-      linear-gradient(var(--vaadin-background-container), var(--vaadin-background-container))
+  [part~='odd-row'] {
+    --vaadin-grid-cell-background-color: var(--vaadin-grid-row-odd-background-color, var(--vaadin-background-color));
+  }
+
+  :host([theme~='row-stripes']) [part~='odd-row'] {
+    --vaadin-grid-cell-background-color: var(
+      --vaadin-grid-row-odd-background-color,
+      color-mix(
+        in srgb,
+        var(--vaadin-text-color) 4%,
+        var(--vaadin-grid-cell-background-color, var(--vaadin-background-color))
```

Comment:
Looks like we only need `var(--vaadin-background-color)` here, as the custom CSS property kind of references itself and then breaks.